contracts/pair_stable_bluna/src/contract.rs

```diff
@@ -0,0 +1,1737 @@
+}
+
+/// ## Description
+/// Stop changing the AMP value. Returns [`Ok`].
+/// ## Params
+/// * **mut config** is the object of type [`Config`].
+///
+/// * **deps** is the object of type [`DepsMut`].
+///
+/// * **env** is the object of type [`Env`].
+fn stop_changing_amp(mut config: Config, deps: DepsMut, env: Env) -> StdResult<()> {
+    let current_amp = compute_current_amp(&config, &env)?;
+    let block_time = env.block.time.seconds();
+
+    config.init_amp = current_amp;
+    config.next_amp = current_amp;
+    config.init_amp_time = block_time;
+    config.next_amp_time = block_time;
+    // now (block_time < next_amp_time) is always False, so we return saved Amp
+
+    CONFIG.save(deps.storage, &config)?;
+
+    Ok(())
+}
+
+/// ## Description
+/// Compute actual amplification coefficient (A)
+/// ## Params
+/// * **config** is the object of type [`Config`].
+///
+/// * **env** is the object of type [`Env`].
+fn compute_current_amp(config: &Config, env: &Env) -> StdResult<u64> {
+    let block_time = env.block.time.seconds();
+
+    if block_time < config.next_amp_time {
+        let elapsed_time =
+            Uint128::from(block_time).checked_sub(Uint128::from(config.init_amp_time))?;
+        let time_range =
+            Uint128::from(config.next_amp_time).checked_sub(Uint128::from(config.init_amp_time))?;
+        let init_amp = Uint128::from(config.init_amp);
+        let next_amp = Uint128::from(config.next_amp);
+
+        if config.next_amp > config.init_amp {
+            let amp_range = next_amp - init_amp;
+            let res = init_amp + (amp_range * elapsed_time).checked_div(time_range)?;
+            Ok(res.u128() as u64)
+        } else {
+            let amp_range = init_amp - next_amp;
+            let res = init_amp - (amp_range * elapsed_time).checked_div(time_range)?;
+            Ok(res.u128() as u64)
+        }
+    } else {
+        Ok(config.next_amp)
+    }
+}
+
+// Bluna reward functions
+
+fn get_bluna_rewarder_instantiating_message(
+    deps: Deps,
+    env: &Env,
+    factory_addr: &Addr,
+) -> Result<SubMsg, ContractError> {
+    Ok(SubMsg {
+        msg: CosmosMsg::Wasm(WasmMsg::Instantiate {
+            admin: None,
+            code_id: query_factory_config(&deps.querier, factory_addr.clone())?.whitelist_code_id,
+            funds: vec![],
+            label: "Bluna rewarder".to_string(),
+            msg: to_binary(&WhitelistInstantiateMsg {
+                admins: vec![env.contract.address.to_string()],
+                mutable: false,
+            })?,
+        }),
+        id: INSTANTIATE_BLUNA_REWARD_HOLDER_REPLY_ID,
+        gas_limit: None,
+        reply_on: ReplyOn::Success,
+    })
+}
+
+fn get_bluna_reward_handling_messages(
+    deps: Deps,
+    env: &Env,
+    info: &MessageInfo,
+    old_user_share: Uint128,
+    old_total_share: Uint128,
+) -> Result<Vec<CosmosMsg>, ContractError> {
+    let mut result: Vec<CosmosMsg> = vec![];
+
+    let config = CONFIG.load(deps.storage)?;
```

Comment:
i think we don't need to load config again, just pass the bluna_rewarder addr instead

Reply:
Yes, you are right